NYPL/fedora_ingest_rails
# [DR-2775] Adding new endpoint for single field updates.

## JIRA tickets:
- [Create New Queue in Fedora Ingest for Single Field Updates](https://jira.nypl.org/browse/DR-2775)

## Things Done:
- Created a new POST endpoint for single field updates. 
- Adjusted some logic assigning time values to fields and relocated some shared methods to a RepoSolrDoc model.

## How to Test:

- Run rspecs
- To make sure nothing is broken, post a normal update to your running local container (I think pointing to dev solr is ok, if you don't have a local solr to point to, as well) via postman. (E.g., `[localhost]/ingest_requests` w/  body `{
    "uuids": ["169880a0-c542-012f-7c43-58d385a7bc34"],
    "test_mode": true
}`
- To test new feature, post a partial update to your running local container via postman. (E.g., `[localhost]/single_field_updates` w/ body `{    "single_field_updates": [ {"uuid": "05286340-cc91-0130-408c-58d385a7b928", "field_name": "vrr_requestable", "field_value": true }, { "uuid": "c4ab1ba0-cc91-0130-4dc3-58d385a7b928", "field_name": "vrr_requestable", "field_value": true }, { "uuid": "borked-uuid-uuid-text", "field_name": "vrr_requestable", "field_value": true } ] }`
- It should create a delayed_job, which you can then invoke through console if you like. Result should be that the first two are successful updates and the third with the invalid uuid is ignored. 
- Test more posts with poorly formatted fields to see some error handling. 

## Future work: 
- We should probably make use of the special queue used by this code by creating a new entry point for background work in AWS. But that can wait. I'll write a followup ticket for that work. For now, it's likely not needed.
